fuel-core/src/database/coin.rs

```diff
@@ -120,4 +123,36 @@ impl Database {
                 == asset_id
         })
     }
+
+    pub fn get_coin_config(&self) -> Result<Option<Vec<CoinConfig>>, anyhow::Error> {
+        let configs = self
+            .iter_all::<Vec<u8>, Word>(COIN, None, None, None)
+            .map(|raw_coin| -> Result<CoinConfig, anyhow::Error> {
+                let coin = raw_coin.unwrap();
+
+                let byte_id = Bytes32::new(coin.0[..32].try_into()?);
+                let output_index = coin.0[32];
+                // Potentially chop off a byte for output index
+                let tx_id = UtxoId::new(byte_id, output_index);
+
+                let ref_coin = Storage::<UtxoId, CoinModel>::get(self, &tx_id)?;
+
+                let ref_coin = ref_coin.unwrap();
```

Comment:
more unwrap

Reply:
This is an option Unwrap not a result unwrap, so should be safe right?

Reply:
Why do we need to fetch the coin by ID if we've already unwrapped it from `raw_coin` above?